src/DurableTask.Netherite/StorageLayer/Faster/PartitionStorage.cs

```diff
@@ -252,21 +260,24 @@ void Check(string workerName, TimeSpan? busySince)
 
         public async Task CleanShutdown(bool takeFinalCheckpoint)
         {
-            this.TraceHelper.FasterProgress("Stopping workers");
-            
-            // in parallel, finish processing log requests and stop processing store requests
-            Task t1 = this.logWorker.PersistAndShutdownAsync();
-            Task t2 = this.storeWorker.CancelAndShutdown();
-
-            // observe exceptions if the clean shutdown is not working correctly
-            await this.TerminationWrapper(t1);
-            await this.TerminationWrapper(t2);
-
-            // if the the settings indicate we want to take a final checkpoint, do so now.
-            if (takeFinalCheckpoint)
+            if (!this.blobManager.ReadonlyMode)
             {
-                this.TraceHelper.FasterProgress("Writing final checkpoint");
-                await this.TerminationWrapper(this.storeWorker.TakeFullCheckpointAsync("final checkpoint").AsTask());
```

Comment:
If we didn't have this if-statement, would our `failIfReadOnly` parameter still help us prevent writes? No, right?

Reply:
No. Inserting this guard was the only way to prevent writes. 

THere is no systematic way to prevent writes as far as I know. Also, systematic prevention would be wrong since there are still some accesses that modify storage (e.g. the leases, and the load publishing).

I am quite sure all writes to application data (partition state) are going through those two calls in PartitionStorage, so I think that is good enough.